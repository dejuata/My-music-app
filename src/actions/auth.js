import {
    getAuth,
    signInWithPopup,
    createUserWithEmailAndPassword,
    updateProfile,
    signInWithEmailAndPassword,
    signOut
} from 'firebase/auth';
import { googleAuthProvider } from '../firebase/firebaseConfig';
import { types } from "../types/types";
import { finishLoading, startLoading } from './ui';
import Swal from 'sweetalert2'


export const startLoginEmailPassword = (email, password) =>{

    return (dispatch) =>{

        dispatch( startLoading() );

        const auth = getAuth();
        signInWithEmailAndPassword(auth, email, password)
            .then(({ user }) => {
                console.log('dispatch login')
                dispatch(login(user.uid, user.displayName, 'firebase', ''))
            })
            .catch( (err) => {
                console.log(err.code, err.message);
                dispatch( finishLoading() );
                Swal.fire('Error', err.message, 'error')
            })

    }
}

export const startRegisterWithEmailPasswordName = (email, password, name) => {
    return (dispatch) => {
        const auth = getAuth();

        createUserWithEmailAndPassword(auth, email, password)
            .then( async ({ user }) => {

                await updateProfile(auth.currentUser, {
                    displayName: name,  photoURL: ""
                })

                dispatch(login(user.uid, user.displayName, 'firebase', ''))

            })
            .catch( (err) => {
                console.log(err.code, err.message);
                Swal.fire('Error', err.message, 'error')
            })

        console.log(auth.currentUser)
    }
}

export const startGoogleLogin = () =>{
    return (dispatch) =>{
        const auth = getAuth();
        signInWithPopup(auth, googleAuthProvider)
            .then( res => {
            })
    }
}

export const login = (uid, name, type, token) => ({
    type: types.login,
    payload: {
        uid,
        name,
        type,
        token
    }
})

export const startLogout = () => {
    return async (dispatch) => {
        const auth = getAuth();
        await signOut(auth)
        dispatch( logout() );
    }
}

export const logout = () => ({
    type: types.logout
})
