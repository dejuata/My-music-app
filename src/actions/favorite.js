import { types } from "../types/types";

export const setFavorites = (payload) => ({
    type: types.setFavorites,
    payload
})

export const removeFavorite= (payload) => ({
    type: types.removeFavorite,
    payload
})