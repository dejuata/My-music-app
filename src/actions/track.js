import { types } from "../types/types";

export const setTracks = (tracks) => ({
    type: types.setTracks,
    payload: tracks
})

export const addTrackFavorite = (id) => ({
    type: types.addTrackFavorite,
    payload: id
})

export const removeTrackFavorite = (id) => ({
    type: types.removeTrackFavorite,
    payload: id
})

export const logoutTrack = () => ({
    type: types.logoutTrack
})