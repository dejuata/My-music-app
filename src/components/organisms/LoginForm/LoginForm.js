import React from 'react';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';

import { useForm } from '../../../hooks/useForm';
import { startLoginEmailPassword } from '../../../actions/auth';

export const LoginForm = () => {

    const dispatch = useDispatch();

    const [ formValues, handleInputChange ] = useForm({
        email: 'demo@gmail.com',
        password: '123456'
    })

    const { email, password } = formValues;

    const handleLogin = (e) => {
        console.log('handleLogin')
        e.preventDefault();
        dispatch( startLoginEmailPassword(email, password) )
    }

    return (
        <form onSubmit={ handleLogin }>
            <div
                className='form-control mb-2'
            >
                <label
                    className='label'
                >
                    Email address
                </label>

                <input
                    type='email'
                    placeholder='example@email.com'
                    name="email"
                    className='input'
                    autoComplete='off'
                    value={ email }
                    onChange={ handleInputChange }
                />
            </div>

            <div
                className='form-control mb-3'
            >
                <label
                    className='label'
                >
                    Password
                </label>

                <input
                    type='password'
                    placeholder='********'
                    name="password"
                    className='input'
                    autoComplete='off'
                    value={ password }
                    onChange={ handleInputChange }
                />
            </div>

            <button
                type='submit'
                className='btn btn-primary'
            >
                Log in
            </button>

            <Link to="/auth/register" className='auth__link'>
                Create new account
            </Link>
        </form>
    )
}
