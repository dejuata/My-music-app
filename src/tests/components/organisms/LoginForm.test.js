import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';
import { Provider } from "react-redux";

import { MemoryRouter } from 'react-router-dom';
import { LoginForm } from '../../../components/organisms/LoginForm/LoginForm';
import { startLoginEmailPassword } from '../../../actions/auth';

// Representar la acción como una función de jest
jest.mock('../../../actions/auth', () => ({
    startLoginEmailPassword: jest.fn()
}))

const middlewares = [ thunk ]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);
// Poder utilizar el dispatch en pruebas
store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={ store }>
        <MemoryRouter>
            <LoginForm />
        </MemoryRouter>
    </Provider>
)

describe('test LoginForm', () => {

    beforeEach( () => {
        store = mockStore(initState);
        // Limpiar mocks
        jest.clearAllMocks();
    })

    test('Should be render LoginForm', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Should be run the action startLoginEmailPassword', () => {

        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });

        expect( startLoginEmailPassword ).toHaveBeenCalledWith('demo@gmail.com', '123456');

    })

})