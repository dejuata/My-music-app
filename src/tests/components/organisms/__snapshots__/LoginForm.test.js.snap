// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LoginPage Should be render LoginForm 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-control mb-2"
  >
    <label
      className="label"
    >
      Email address
    </label>
    <input
      autoComplete="off"
      className="input"
      name="email"
      onChange={[Function]}
      placeholder="example@email.com"
      type="email"
      value="demo@gmail.com"
    />
  </div>
  <div
    className="form-control mb-3"
  >
    <label
      className="label"
    >
      Password
    </label>
    <input
      autoComplete="off"
      className="input"
      name="password"
      onChange={[Function]}
      placeholder="********"
      type="password"
      value="123456"
    />
  </div>
  <button
    className="btn btn-primary"
    type="submit"
  >
    Log in
  </button>
  <a
    className="auth__link"
    href="/auth/register"
    onClick={[Function]}
  >
    Create new account
  </a>
</form>
`;
